test/topology/test_start_bootstrapped_with_invalid_seed.py

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import pytest
+import time
+import logging
+
+from test.pylib.internal_types import IPAddress
+from test.pylib.manager_client import ManagerClient
+from test.pylib.util import wait_for_cql_and_get_hosts, wait_for_cql
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail(reason="issue #14945")
+async def test_raft_snapshot_truncation(manager: ManagerClient):
+    """
+    Check that when bootstrapped, having a non-resolving DNS name in
+    the config, does prevent node restart.
+    Issue https://github.com/scylladb/scylladb/issues/14945.
+
+     1. Start 3 nodes.
+     2. Wait till all start and join the cluster.
+     3. Stop one node & and start it with a non-existing name in the seeds list config.
+     4. Make sure the node started successfully since it's already a cluster member.
+
+    """
+    servers = await manager.servers_add(3)
+    cql = manager.get_cql()
+
+    s1 = servers[0]
+    await wait_for_cql_and_get_hosts(cql, [s1], time.time() + 60)
+
+    await manager.server_stop_gracefully(s1.server_id)
+
+    await manager.server_start(s1.server_id, seeds=[IPAddress("no_address")], wait_interval=20)
```

Comment:
@bitpathfinder I recall we discussed on 1on1 that the PR should also include a regression test -- to check that seed resolution still works when a correct DNS name is provided. A regression test  for seed resolution code would be desirable since you're modifying the code.

Reply:
Ok, will add a regression.

Reply:
Don't worry, when I write 'regression', I mean a test that checks that there is no functionality regression. I can't get rid of that habit from the previous place.